[Add] navbar burger icon

kimchirice opened the pull request:
## Rationale

Add hamburger icon for mobile menu display


## Advice for Reviewers & Testing Notes

- You should be able to see Burger icon on mobile screen
- Burger icon onClick event should trigger NavLink
- Previous create file *_/**src/compnents/NavBurger.vue**_* renamed to ***_/src/components/NavBarMobile.vue_***

## Screenshots:

- ![Screenshot 2020-07-21 00 30 25](https://user-images.githubusercontent.com/36904351/87950298-760ddf00-caea-11ea-83bd-485d1dae3279.png)


## Task Name 
- Navigation hamburger [#21](https://github.com/codesydney/codesydney/issues/21)



## Linting Checklist
- [x] No commented code
- [x] Code Formatted nicely (Prettier)
- [x] PR your own code before you assign reviewers

QjStout commented on client/src/App.vue:
```diff
@@ -8,13 +8,89 @@
+};
 </script>
 
-<style scoped>
```
Why have you removed the scoping?
Let me know if I'm wrong, but won't that potentially create conflicts with stylesheets from other pages?

QjStout replied:
After discussing with @kimchirice, turns out that the app level style sheet is not scoped by default. Closing this change request.

QjStout commented on client/src/App.vue:
```diff
@@ -8,13 +8,89 @@
+      <div id="nav">
+        <div
+          id="navBar-icon"
+          v-if="mobileView"
+          @click="showNavBar = !showNavBar"
```
@kimchirice, we decided on using the hamburger for both mobile & web.
Sorry to throw out your extra work.

QjStout replied:
We are using https://bit.ly/3fNB6iV as a reference.
Obviously I'm not expecting you to spend hours making all that complicated animation though.

The main downside of the burger is low click-through, but the idea is that we make the homepage cover everything needed for clients and the other pages are really just for Code.Sydney members.

QjStout commented on client/src/components/NavBarMobile.vue:
```diff
@@ -0,0 +1,37 @@
+  <div id="navBar-mobile">
+    <!-- placeholder Nav links -->
+    <ul>
+      <li>Home</li>
+      <li>Contact</li>
```
```suggestion
      <router-link to="/">Home</router-link>
      <router-link to="/contact">Contact</router-link>
```

QjStout commented on client/src/components/NavLink.vue:
```diff
@@ -0,0 +1,20 @@
+<template>
+  <div>
+    <router-link to="/">Home</router-link> |
+    <router-link to="/contact">Contact</router-link>
+  </div>
+</template>
+
+<script>
+export default {};
+</script>
+
+<style scoped>
+a.router-link-exact-active {
+  color: #42b983;
+}
+a {
+  font-weight: bold;
+  color: #2c3e50;
+}
+</style>
```
Remove the NavLink. We are going to use burger for mobile and web.
It's a cleaner look and we don't care about the lower click-through rates.

promie commented on client/src/App.vue:
```diff
@@ -8,13 +8,89 @@
+  },
+  methods: {
+    toggleMobileView() {
+      this.mobileView = window.innerWidth <= 990;
```
Not sure how i feel about this. Should we be doing proper media query?

QjStout replied:
I don't know what best practice is here.
You guys let me know what you decide.

kimchirice commented:
hi all,  need help with this, @QjStout @promie @chenhe-henry @engramar 

me trying to implement [this solution](https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_overlay2) 
there's [another similar one](https://codepen.io/kimchirice/pen/ExPrGrV) @chenhe-henry sent to me this am,


but stuck, with how to toggle div height from 0% to 100% in Vue.js, I created 2 class, 'navbar' and 'navbar_open'
```Vue
`<template>
  <div id="app">
    <!-- Banner/Header -->
    <!-- Hamburger -->
    <div id="myNav" class="navbar" :class="{ navbar_open: showNavbar }">
      <div class="closebtn" v-on:click="showNavbar = !showNavbar">
        <i class="fa fa-times"></i>
      </div>

      <div class="navbar-content">
        <a>Home</a>
        <a>Contact</a>
      </div>
    </div>

    <div class="openbtn">
      <i class="fa fa-bars" v-on:click="showNavbar = !showNavbar"></i>
    </div>

    <router-view />
  </div>
</template>

<script>
export default {
  data() {
    return {
      showNavbar: false
    };
  },
  methods: {
    toggleNavbar() {
      this.showNavbar = !showNavbar;
    },
    setHeightValue(showNavbar) {
      const heightValue = this.showNavbar ? '100%' : '0%';
      return heightValue;
    }
  },
  created() {
    this.setHeightValue();
  }
};
</script>

<style>
@import url('https://use.fontawesome.com/releases/v5.9.0/css/all.css');

.navbar_open {
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.9);
  overflow-y: hidden;
  transition: 0.5s;
  background-color: #2b4c72;
}

.navbar {
  width: 100%;
  height: 0%;
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.9);
  overflow-y: hidden;
  transition: 0.5s;
  background-color: #2b4c72;
}

.navbar-content {
  position: relative;
  top: 25%;
  width: 100%;
  text-align: center;
  margin-top: 30px;
}

.navbar a {
  padding: 8px;
  text-decoration: none;
  font-size: 36px;
  color: #818181;
  display: block;
  transition: 0.3s;
}

.navbar a:hover,
.navbar a:focus {
  color: #f1f1f1;
}

.navbar .closebtn {
  position: absolute;
  top: 20px;
  left: 45px;
  font-size: 60px;
  color: white;
  cursor: pointer;
}

.openbtn {
  font-size: 30px;
}

@media screen and (max-height: 450px) {
  .navbar {
    overflow-y: auto;
  }
  .navbar a {
    font-size: 20px;
  }
  .navbar .closebtn {
    font-size: 40px;
    top: 15px;
    right: 35px;
  }
}
</style>
```

kimchirice commented:
[Add] a new PR #50
